### ISSUE TITLE
Option to disable hover widget

### ISSUE THREAD

**Issue by osher** [microsoft/vscode#30114]
- VSCode Version: 1.13.1
- OS Version: win10

Steps to Reproduce:

1. open a JS file
2. hover a param

Expect: NOTHING

Found
![image](https://user-images.githubusercontent.com/803101/27853658-3bfdd292-616c-11e7-9a78-6c54a919a008.png)

I ***REALLY*** want it to stop doing that! I want vs-code to be the cool minimalist tool I started with, and it gets bloated after every update

**Comment by osher** [comment]
After googling - I found 
```
  "typescript.validate.enable": false,
```
which is evidently not it.
Couldn't find anything else, just debates over the same setting.
Please help?

**Comment by osher** [comment]
That's all the settings I could find, none of them do the trick:

```
  "typescript.disableAutomaticTypeAcquisition": true,
  "typescript.useCodeSnippetsOnMethodSuggest": false,
  "typescript.check.npmIsInstalled": false,
  "typescript.check.tscVersion" : false,
  "typescript.validate.enable": false,
  "typescript.format.enable": false,
  "typescript.referencesCodeLens.enabled": false,
 ```

Please help... it bothers me so much I 'll just have to stop using vscode

**Comment by joaomoreno** [comment]
```
"editor.parameterHints": false
```

**Comment by osher** [comment]
Doesn't work :(

**Comment by osher** [comment]
```
// Place your settings in this file to overwrite the default settings
{ "editor.fontSize": 18,
  "editor.wordWrap": "off",
  "editor.wrappingIndent": "indent",
  "search.exclude": {
		"**/node_modules": false      
  },
  "editor.parameterHints": false,
  "typescript.disableAutomaticTypeAcquisition": true,
  "typescript.useCodeSnippetsOnMethodSuggest": false,
  "typescript.check.npmIsInstalled": false,
  "typescript.check.tscVersion" : false,
  "typescript.validate.enable": false,
  "typescript.format.enable": false,
  "typescript.referencesCodeLens.enabled": false,
  "workbench.colorTheme": "Visual Studio Light"
}
```

![image](https://user-images.githubusercontent.com/803101/27856505-0cce2764-6177-11e7-87fa-a5e711cb6181.png)

**Comment by osher** [comment]
@joaomoreno  - please reopen...

**Comment by osher** [comment]
went over all the settings, disabled anything that might make sense, for no avail :(

even tried restarting the app
```
{ "editor.fontSize": 18,
  "editor.wordWrap": "off",
  "editor.wrappingIndent": "indent",
  "search.exclude": {
		"**/node_modules": false      
  },
  "editor.parameterHints": false,
  "typescript.disableAutomaticTypeAcquisition": true,
  "typescript.useCodeSnippetsOnMethodSuggest": false,
  "typescript.check.npmIsInstalled": false,
  "typescript.check.tscVersion" : false,
  "typescript.validate.enable": false,
  "typescript.format.enable": false,
  "typescript.referencesCodeLens.enabled": false,
  "workbench.colorTheme": "Visual Studio Light",
  "editor.quickSuggestions": {
    "other": false,
    "comments": false,
    "strings": false
  },
  "editor.autoClosingBrackets": false,
  "editor.snippetSuggestions": "none"
}
```

Help?

**Comment by joaomoreno** [comment]
I guess you want a setting to disable the **hover** widget.

**Comment by osher** [comment]
ok, if that's what it's called...

I don't want the editor to popup anything unless I did something to trigger it. 
Hovering does not count, because I move the mouse while I think, and the popups break my train of thoughts.

man, this actually ended up costing me money...

**Comment by osher** [comment]
tl;dr:
consider a mode that this popup appears in response to a triggered user action other than hover.
something in the spirit of Ctrl+shift+space, which resembles Ctrl+space for auto-completion
or double/tripple click for mouse users 

:)

**Comment by alexandrudima** [comment]
There is a "hidden" setting, `"editor.hover": false`, which entirely disables the hover widget. It is not public, nor documented, as there are some complicated interactions of this setting with debugging. e.g. the setting will be reset to true after a debug session...

**Comment by osher** [comment]
wow. it worked!   **what a relief...**
It did require restart of the editor though (and since the motto is working with editor instance per git repo - they all had to be restart... 😛 )
You say it resets after a debug session? I rarely debug - I usually run with prints, however - If I'll see it jumps - I'll update with my experience.

**Comment by osher** [comment]
Anyway - to make a close of this issue, what I believe we should do is:
 - document the current status
 - make it stay between debug sessions (if it still jumps)
 - allow configure a way to invoke it by a shortcut-key stroke of the user initiative
 - document the end result

If you think it needs to be handled in few different issues - I'll adhere, just let me follow them too 👍

**Comment by stepankuzmin** [comment]
Hi there! Is there any way to disable typescript typings on hover? E.g. I'm using https://github.com/flowtype/flow-for-vscode and want to see only flow typings, so disabling hover completely is not the solution.

<img width="440" alt="screenshot 2018-02-27 14 43 24" src="https://user-images.githubusercontent.com/533564/36727067-f1f1cbec-1bcc-11e8-8cdd-30191aacde78.png">

**Comment by wolfscott** [comment]
I like the "editor.hover": false setting (thanks tons, @alexandrudima), but not that I have to bounce the app to get it to take effect because sometimes I want them, sometimes I don't - mounds -vs- almond joy 

suggestion would be to have a "hot key" to toggle it on/off - and personally I wouldn't care if it didn't survive a restart and default "enabled" if I could just hit a short-cut to disable every popup at-will without bouncing the app.  "Silent" mode as it were.

**Comment by audiot** [comment]
manually activate the tooltips / colorpicker:
--------------------------------------------

if you have set "editor.hover" to false you can manually activate the tooltips or the colorpicker by pressing the shortcut  "CTRL+K+I". You can close the tooltip with "ESC" (if the tooltip has lost the focus, just click on the tooltip again and press "ESC")

Thank me later :)

**Comment by wolfscott** [comment]
@audiot, that's a great tip;  single-use toggle  -- many thanks!! (it's later) :)

**Comment by celltowertrees** [comment]
Hi there,

I'm using VSCode 1.25.1 on OSX 10.11.6 El Capitan. This exact issue is happening to me except setting editor.hover to false does nothing. I'm at the end of my rope here, I do not need giant tooltips hovering in my face explaining what a `<div>` element is.

I've set
```
"editor.hover": false,
"editor.parameterHints": false,
"editor.quickSuggestions:" false,
```
and I still get the hover tooltips, everywhere.

Is there anything else I can do? Does this option no longer exist?

**Comment by alexandrudima** [comment]
Yes, the option is now `"editor.hover.enabled": false`.

**Comment by Nathan-Brandless** [comment]
`editor.hover.enabled` seems to not work if you also have `editor.hover` defined. I had to delete the old line before it would honor the new one.

**Comment by silvenon** [comment]
This technically resolves this issue, but what many of us really want is the option to remove TypeScript types from the hover widget. It's really confusing when using the Flow extension. https://github.com/Microsoft/vscode/issues/30114#issuecomment-368848010

**Comment by thecodecafe** [comment]
Have been looking for a way to disable the hover widget for a long time now, I actually stopped using VSCode because of that, done it but sadly this disables every pop up even when you intentionally hold down the command key, you don't get the inteli-sense,  the main reason why anyone would want to remove the pop up is to avoid distractions when moving their mouse across their code, but intentionally holding down command is a different thing altogether, when a user does that then they really want to display some inteli-sense info.

**Comment by chasebank** [comment]
Yeah this took way too much time. There's too many synonymous features that all sound like they may affect this (ie: "suggestions", "definitions", etc) and way too many settings surrounding them, that lead to a rabbit hole of Google results that frustratingly don't solve the problem.

This setting needs to be a clearly labeled checkbox in the main Text Editor settings.

**Comment by istarkov** [comment]
@stepankuzmin Did u found a solution to disable non flow typings?

**Comment by siteit** [comment]
I think a smart way is to allow "turning off these notification" on the popover itself. it's crazy annoying.

**Comment by bobanum** [comment]
I can't believe having to ask for an OPTION to turn popups off.  Nobody needs them. Remove them altogether and I'll make an "Annoying popups" extension for those wanting to distract their kittens. 18 months and this issue is still opened.

**Comment by audiot** [comment]
I made an extension, that switches the hover-widget on and off:

https://marketplace.visualstudio.com/items?itemName=RoyAction.toggle-hover

... read the description!

**Comment by dominicfraser** [comment]
@istarkov 
> @stepankuzmin Did u found a solution to disable non flow typings?

I turned these off using: Extensions -> search `@builtin types` -> disable `TypeScript and Javascript Language Features`

**Comment by webappslove** [comment]
Found `editor.hover.delay` was perfect for me, the problem is not that I don't want the hover widget, but that it was always blocking vision and clicks.

**Comment by david-bressler** [comment]
I still can't get rid of these very annoying popups. Every time I use my python debugger, when I hover over a variable name it pops up a huge popup showing the content of the variable, which is very distracting and hurts my eyeballs. I've tried everything in this and other threads to no avail. Here's my settings.json: 
```

{
  "editor.hover.enabled": false,
  "editor.hover.sticky": false,
  "editor.quickSuggestions": {
      "other": false,
      "comments": false,
      "strings": false
    },
    "editor.parameterHints.enabled": false,
    "editor.suggest.snippetsPreventQuickSuggestions": false,
    "html.suggest.html5": false,
    "editor.snippetSuggestions": "none",
    "editor.codeLens": false,
    "typescript.validate.enable": false,
    "editor.hover.delay": 300000,
    
}
```
Some other info: I'm using VSCode as a remote editor, so don't know if that has an impact here. Can anyone help? I'll likely abandon VSCode if I can't resolve this.

**Comment by bobanum** [comment]
We shouldn't HAVE to disable it, it shouldn't just be there. From a UI design point of view, unsollicited clickable and selectable and HUGE popups are a abysmally bad choice.
I don't have the editor choice in my environment... sadly.

**Comment by octref** [comment]
Putting together all info here:

- `"editor.hover.enabled": false` if you want the hover widget gone completely
- Set `editor.hover.delay` if you want a longer delay between hovering and showing the widget
- If you just want TypeScript's hover gone, type `@builtin typescript` into extension widget and disable it. Beware this will disable all TypeScript language features, and you may need another language server such as flow to get auto-completion etc

OP asked for an option, and there's an option now, thus closing.

**Comment by chasebank** [comment]
Considering his issue was opened over 2 years ago, and there is an extremely excessive number of "hover" and "tooltip" settings, I really think this needs to be added to the Text Editor Settings in the UI, which is where a user would logically expect it to be.

The bulk of this thread (including OP) is centered around the terrible UX related to this setting, and the current implementation does not address that.

Issue should be re-opened.

**Comment by octref** [comment]
> I really think this needs to be added to the Text Editor Settings in the UI, which is where a user would logically expect it to be.

If you have concrete suggestions, feel free to open a new issue.

> terrible UX

If you have concrete cases when UX is terrible (include a screenshot / screen recording if possible), open new issues.

**Comment by bobanum** [comment]
Here's an example of terrrible UX : 
![Terrible UX example](https://user-images.githubusercontent.com/32491185/67518060-f2fe4c00-f671-11e9-8a89-e40bc639e20b.png)
Who needs to be reminded all the time, not on a click event or a keypress, but a mouseover, of the definition of the li tag? Moreover, wanting to click in the line above fires a link.
I would be glad to be part of a general discussion about the general UX of VS Code. We have the proof here, 2 years later, that opening an issue won't do anything.

**Comment by octref** [comment]
If you don't like it, you can turn it off with the option `"editor.hover.enabled": false`. You can also increase the delay so it doesn't show unless you hover on it for 3 seconds.

While you think it is not useful to you, keep in mind that:
- There are many users who are new to HTML and such hover can be useful for them.
- There are new or less-used HTML elements that little people know about. Off the top of your head, can you tell me what are `<rt>`, `<bdi>` or `<wbr>`?
- If you are working on a language you are less familiar with, you might need these hover information as well

**Comment by david-bressler** [comment]
> If you don't like it, you can turn it off with the option `"editor.hover.enabled": false`. You can also increase the delay so it doesn't show unless you hover on it for 3 seconds.
> 
> While you think it is not useful to you, keep in mind that:
> 
> * There are many users who are new to HTML and such hover can be useful for them.
> * There are new or less-used HTML elements that little people know about. Off the top of your head, can you tell me what are `<rt>`, `<bdi>` or `<wbr>`?
> * If you are working on a language you are less familiar with, you might need these hover information as well

As I posted above, "editor.hover.enabled": false does NOT work.

**Comment by istarkov** [comment]
Yep, this https://github.com/microsoft/vscode/issues/30114#issuecomment-368848010 is the main issue. As a user of flowtype I want to see only flowtype hints at js files (at least with @flow annotation) I don't want to disable all hovers, I don;t want to disable ts for all files. And delay id perfect for me. So issue can't be closed as it's not solved somehow

**Comment by octref** [comment]
> As I posted above, "editor.hover.enabled": false does NOT work.

As explained above two years ago: https://github.com/microsoft/vscode/issues/30114#issuecomment-321490156

![image](https://user-images.githubusercontent.com/4033249/67528131-8ed98900-f66d-11e9-84b8-658f755a6664.png)

> As a user of flowtype I want to see only flowtype hints at js files (at least with @flow annotation) I don't want to disable all hovers, I don;t want to disable ts for all files. And delay is perfect for me. So issue can't be closed as it's not solved somehow

That's all you can do now: Either use TS or Flow

![image](https://user-images.githubusercontent.com/4033249/67528170-a4e74980-f66d-11e9-84cb-3b33b09a8076.png)

You are saying you don't like TS and Flow conflicting with each other, and you want features from both. There's no way to do that, and that's not the same as the original issue.

**Comment by david-bressler** [comment]
It's ok I've completely abandoned the vscode debugger in favor of pdb. I encourage others to do the same.
